### PR TITLE
Enables "claiming" of NFT-did

### PR DESCRIPTION
**Pull request by trepca**
When user accepts an offer for NFT, they will receive it without any DID set, so we need this function to set DID. 

New RPC api `nft_set_nft_did`

### CODE REVIEW

File: tests/wallet/nft_wallet/test_nft_wallet.py [hunk 1]
```diff
@@ -808,3 +808,136 @@ async def test_nft_rpc_mint(two_wallet_nodes: Any, trusted: Any) -> None:
...
+    assert compute_memos(sb)
+    await time_out_assert_not_none(5, full_node_api.full_node.mempool_manager.get_spendbundle, sb.name())
+
+    for i in range(1, num_blocks):
+        await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
+    await time_out_assert(10, wallet_0.get_unconfirmed_balance, 13999999999998)
+    await time_out_assert(10, wallet_0.get_confirmed_balance, 13999999999998)
+    # Check DID NFT
+    time_left = 5.0
+    while time_left > 0:
```
Comment (trepca) [inline tests/wallet/nft_wallet/test_nft_wallet.py:930]: I'll be cleaning test code and make it more reusable in separate PR

File: chia/wallet/nft_wallet/nft_wallet.py [hunk 2]
```diff
@@ -628,6 +634,30 @@ async def transfer_nft(
         self.wallet_state_manager.state_changed("nft_coin_transferred", self.wallet_info.id)
         return nft_tx_record.spend_bundle
 
+    async def set_nft_did(self, nft_coin_info: NFTCoinInfo, did_id: bytes32, fee: uint64 = uint64(0)) -> SpendBundle:
+        self.log.info("Setting NFT DID with parameters: nft=%s did=%s", nft_coin_info, did_id)
+        unft = UncurriedNFT.uncurry(nft_coin_info.full_puzzle)
```
Comment (ytx1991) [inline chia/wallet/nft_wallet/nft_wallet.py:639]: Type def for variables

File: chia/wallet/wallet_state_manager.py [hunk 3]
```diff
@@ -716,8 +716,14 @@ async def handle_nft(
         """
         wallet_id = None
         wallet_type = None
-        did_id = uncurried_nft.owner_did
-        self.log.debug("Handling NFT: %s， DID: %s", coin_spend, did_id)
+        did_id = None
+        if uncurried_nft.supports_did:
+            # extract the latest did_id from solution
+            solution = coin_spend.solution.to_program()
+            did_id = solution.at("rrfffrfrrfrf").atom
```
Comment (ytx1991) [inline chia/wallet/wallet_state_manager.py:723]: Better to create a util method for this

**Comment by trepca** [comment]
needs to be rewritten to use generated signed tx, waiting for #11899

**Pull request closed without merging**